infra/gcp/tests/gke.tf

```diff
@@ -105,6 +105,10 @@ module "prow_build" {
   filestore_csi_driver       = false
   create_service_account     = false
   remove_default_node_pool   = true
+  cluster_autoscaling = {
```

Comment:
Can you remove this?
https://cloud.google.com/kubernetes-engine/docs/reference/rest/v1beta1/projects.locations.clusters#clusterautoscaling

This is already enabled in a different way. 

https://cloud.google.com/kubernetes-engine/docs/concepts/cluster-autoscaler#how_cluster_autoscaler_works

Autoscaling on GKE can done by either:
- Enabling autoscaling on a nodepool - This adds an extra instance to the existing nodepool. Faster as the GCE MIG is already there and as well the nodepool. We have sane limits for now. The cap is 8 instances but we can raise that instead.
- Enabling autoscaling on a cluster - This spins up new nodepools when the cluster needs more resources. Slow as GKE creates a nodepool and the backing GCE infrastructure. An extra config to write

Reply:
Probably want to capture this (if it is not already) somewhere in the `gcp/` folder @upodroid .

Reply:
@upodroid the main thing I want is the `autoscaling_profile = "OPTIMIZE_UTILIZATION"` here which enables scaling down the cluster more aggressively - https://cloud.google.com/kubernetes-engine/docs/concepts/cluster-autoscaler#autoscaling_profiles. I don't think the cluster-level autoscaling config is **only** for spinning up new nodepools.

Reply:
@upodroid another further thoughts on this =) ? Is it okay to merge now?

Reply:
This is on hold for a bit longer.

Current diff is failing with the following error.

```
 REDACTED  MCW0CDP3YY  ~  Desktop  …  kn-test-infra  infra  gcp   move-last-custom-jobs  1✎  5⚑  $    terraform apply -target module.tests.module.prow_build
╷
│ Error: Invalid value for module argument
│ 
│   on tests/gke.tf line 108, in module "prow_build":
│  108:   cluster_autoscaling = {
│  109:     enabled = true
│  110:     autoscaling_profile = "OPTIMIZE_UTILIZATION"
│  111:   }
│ 
│ The given value is not suitable for child module variable "cluster_autoscaling" defined at
│ .terraform/modules/tests.prow_build/variables.tf:179,1-31: attributes "gpu_resources", "max_cpu_cores", "max_memory_gb", "min_cpu_cores",
│ and "min_memory_gb" are required.
╵

```

We need to switch to the beta submodule

So apply the following diff:

```
 REDACTED  MCW0CDP3YY  ~  Desktop  …  kn-test-infra  infra  gcp   move-last-custom-jobs  2✎  5⚑  ERROR  $    git diff
diff --git a/infra/gcp/tests/gke.tf b/infra/gcp/tests/gke.tf
index 8f6af499..b53dde17 100644
--- a/infra/gcp/tests/gke.tf
+++ b/infra/gcp/tests/gke.tf
@@ -90,7 +90,7 @@ module "prow" {
 }
 
 module "prow_build" {
-  source                     = "terraform-google-modules/kubernetes-engine/google"
+  source                     = "terraform-google-modules/kubernetes-engine/google//modules/beta-public-cluster"
   project_id                 = module.project.project_id
   name                       = "prow-build"
   region                     = "us-central1"
@@ -105,6 +105,17 @@ module "prow_build" {
   filestore_csi_driver       = false
   create_service_account     = false
   remove_default_node_pool   = true
+  gce_pd_csi_driver          = true
+  cluster_autoscaling = {
+    enabled             = false
+    autoscaling_profile = "OPTIMIZE_UTILIZATION"
+    gpu_resources       = []
+    max_cpu_cores       = null
+    max_memory_gb       = null
+    min_cpu_cores       = null
+    min_memory_gb       = null
+
+  }
 
```
`

Reply:
Thanks for checking on this.
Do we have to add the empty fields? Also shouldn't `cluster_autoscaling.enabled` be `true`?

Reply:
- Empty variables: yes it is a terraform limitation/feature 
- setting enabled to false: This is correct, that block creates nodepool auto provisioning which we don't want but we want to set the Autoscaling profile.

Reply:
Thanks for the clarification. Done.